### PR TITLE
Add `chainerx.erf`

### PR DESCRIPTION
**Pull request by aksub99**
Add `chainerx.erf`

### CODE REVIEW

**Comment by aksub99** [comment]
This PR is ready for review:)

File: chainerx_cc/chainerx/routines/explog.cc [hunk 1]
```diff
@@ -14,10 +14,34 @@
...
 #include "chainerx/kernels/explog.h"
 #include "chainerx/routines/creation.h"
+#include "chainerx/routines/math.h"
+#include "chainerx/routines/misc.h"
 #include "chainerx/routines/type_util.h"
 
 namespace chainerx {
 
+Array Erf(const Array& x) {
+    const float pi = std::acos(-1);
```
Comment (niboshi) [inline chainerx_cc/chainerx/routines/explog.cc]: Let's define PI in `constant.h` in this way:

```
constexpr long double kPi = 3.141592653589793238462643383279502884L;
```

File: chainerx_cc/chainerx/routines/explog.cc [hunk 2]
```diff
@@ -14,10 +14,34 @@
...
+    {
+        NoBackpropModeScope scope{};
+        x.device().backend().CallKernel<ErfKernel>(x, out);
+    }
+
+    BackwardBuilder bb{"erf", x, out};
+    if (BackwardBuilder::Target bt = bb.CreateTarget(0)) {
+        bt.Define([x_tok = bb.RetainInput(0), pi](BackwardContext& bctx) {
+            const Array& x = bctx.GetRetainedInput(x_tok);
+            bctx.input_grad() = *bctx.output_grad() * 2 / std::pow(pi, 0.5) * Exp(-Square(x));
```
Comment (niboshi) [inline chainerx_cc/chainerx/routines/explog.cc]: `std::pow(pi, 0.5)` could be written as `std::sqrt(pi)` (more efficient/readable).

Comment (aksub99) [reply chainerx_cc/chainerx/routines/explog.cc]: Sure! Thanks.

**Comment by niboshi** [comment]
Thanks!
Jenkins, test this please

**Comment by aksub99** [comment]
The errors seem to be caused by `batch_normalization`. They are unrelated to this PR right?

**Comment by niboshi** [comment]
Yes, that's correct.
Thank you so much!

**Pull request merged**